### PR TITLE
Refactor Code to Remove Deprecated ioutil Package and Improve Formatting

### PR DESCRIPTION
**Pull request by Tom5521**
- Replaced usages of the deprecated `ioutil` package with the appropriate `os` and `io` functions (`os.ReadFile`, `os.WriteFile`, `os.CreateTemp`, `io.ReadAll`).
- Updated file permission syntax to octal notation.

### CODE REVIEW

File: apply_test.go [hunk 1]
```diff
@@ -129,7 +128,58 @@ func TestCorruptPatch(t *testing.T) {
...
+		0xa6,
+		0xd6,
+		0xb2,
+		0x1b,
+		0xfb,
+		0xe6,
+		0xbe,
+		0x7b,
+		0x70,
+	}
```
Comment (ccoVeille) [inline apply_test.go:182]: This change doesn't seem necessary.

Did you apply gofumpt on files?

I'm curious how you get this

File: hide_windows.go [hunk 2]
```diff
@@ -9,7 +9,12 @@ func hideFile(path string) error {
 	kernel32 := syscall.NewLazyDLL("kernel32.dll")
 	setFileAttributes := kernel32.NewProc("SetFileAttributesW")
 
-	r1, _, err := setFileAttributes.Call(uintptr(unsafe.Pointer(syscall.StringToUTF16Ptr(path))), 2)
+	utf16, err := syscall.UTF16PtrFromString(path)
+	if err != nil {
+		return err
+	}
+
+	r1, _, err := setFileAttributes.Call(uintptr(unsafe.Pointer(utf16)), 2)
```
Comment (ccoVeille) [inline hide_windows.go:17]: This change is not related to what is described in the PR description

I would have put in away from this PR and moved to another one, dedicated to this change.

**Comment by ccoVeille** [comment]
Except the few comments I raised, thanks for the PR

File: apply_test.go (hunk 1 above)
Comment (Tom5521) [reply apply_test.go:182]: > Did you apply gofumpt on files?

I honestly don't know. I also had golines enabled in my editor, so that's probably what it was.

File: hide_windows.go (hunk 2 above)
Comment (Tom5521) [reply hide_windows.go:17]: Oh, sorry about that. I accidentally overlooked it

Comment (ccoVeille) [reply hide_windows.go:17]: Thanks for confirming it.

I'll mark the PR as "changes required"

**Review by ccoVeille — CHANGES_REQUESTED** [review]
Please address

https://github.com/minio/selfupdate/pull/15#discussion_r1831662643

https://github.com/minio/selfupdate/pull/15#discussion_r1831664761

File: apply_test.go (hunk 1 above)
Comment (ccoVeille) [reply apply_test.go:182]: As you may update your PR for [this](https://github.com/minio/selfupdate/pull/15#discussion_r1831664761), you could also revert this
